### PR TITLE
210 history bug at app

### LINKED ISSUE #210: History Bug at `/app`
To recreate: 
- Have courses stored in `localStorage` (upload something)
- Exit the page and reload it at `/` (landing page)
- Click "get started" button
- You are now on `/app`. Click the back arrow of your browser

Expected behaviour: return to landing page `/`
Actual behaviour: have to click 3 times to return to landing page `/`

Probably something to do with with 
https://github.com/techstartucalgary/lifeline/blob/ca442d8c06f18b58a9b38d2430ecea64d1f5ac07/client/src/pages/Review/Review.tsx#L96-L103
interacting with https://github.com/techstartucalgary/lifeline/blob/ca442d8c06f18b58a9b38d2430ecea64d1f5ac07/client/src/pages/Review/Review.tsx#L69-L77

as I think it was introduced in #202 but I'm not sure.

### PR DESCRIPTION
**Pull request by cloudyyoung**
This PR closes #210.

This PR also introduces two minor issues:
1. NavigationPanel renders slowly that needs performance optimization
2. Mobile screen iOS, page transition animation should be omitted with gesture

https://user-images.githubusercontent.com/20573623/226564826-67566c4f-1057-4187-8889-49ae050622b1.mov

https://user-images.githubusercontent.com/20573623/226565134-eecde30c-1a7f-424e-ad2c-e5644a435494.mp4

### CODE REVIEW

File: client/src/pages/Review/Review.tsx
Comment (tim-macphail) [inline client/src/pages/Review/Review.tsx:1]: So does `useNavigate()` automatically use the history API?

**Comment by tim-macphail** [comment]
> 1. NavigationPanel renders slowly that needs performance optimization
I don't encounter this, can you explain more?

**Comment by cloudyyoung** [comment]
> > 1. NavigationPanel renders slowly that needs performance optimization
> 
> I don't encounter this, can you explain more?

So if you look at the video in the PR description, do you see that the NavigationDrawer highlight switches a bit too slow? 
The course info panel almost changes immediately when I click, but the nav drawer highlight shifted about ~500ms after which feels laggy.

**Comment by tim-macphail** [comment]
> > > 1. NavigationPanel renders slowly that needs performance optimization
> > 
> > 
> > I don't encounter this, can you explain more?
> 
> So if you look at the video in the PR description, do you see that the NavigationDrawer highlight switches a bit too slow? The course info panel almost changes immediately when I click, but the nav drawer highlight shifted about ~500ms after which feels laggy.

Ok I see. Any idea the cause of this? In the mean time I think it's worth the trade off of having history working properly.

**Review by tim-macphail — APPROVED** [review]
Great work!

**Pull request merged**